shared/src/Twin.cs

```diff
@@ -147,9 +147,17 @@ public Twin(TwinProperties twinProperties)
         /// </summary>
         /// <param name="formatting">Optional. Formatting for the output JSON string.</param>
         /// <returns>JSON string</returns>
-        public string ToJson(Formatting formatting = Formatting.None)
+        public string ToJson(Formatting formatting)
         {
             return JsonConvert.SerializeObject(this, formatting);
         }
+        /// <summary>
```

Comment:
Looking at the issue details from #1024 , why are we modifying the SDK to be compatible with 9.0.1, instead of having the referencing application use the updated version of NewtonSoft json. That way, the responsibility of backward-compatibility rests on NewtonSoft json directly.

Reply:
Okay, I'll re-open the other PR and just let this PR focus on upgrading the existing dependency